### DISCUSSION TITLE
Is there a way to print a quick static string while the app binary initializes itself?

### DISCUSSION THREAD

**Discussion by zapta** [pyinstaller/pyinstaller#7509] [Help]
I created a single file app using pyinstaller and it works very well, thanks.

The only problem is that there is no feedback in the first 5-10 seconds, before the actual app starts. I saw references to a splash screen feature but I wonder, is there a way to print immediate a user provider static string (text, in the console)? E.g. just identifying the program and letting the user know what it will take a few seconds to start.

Alternatively, does pyinstaller provide any use feedback during the app initialization such as a progress bar or any other textual indication?

**Comment by rokm** [comment]
> I created a single file app using pyinstaller and it works very well, thanks.
> 
> The only problem is that there is no feedback in the first 5-10 seconds, before the actual app starts. I saw references to a splash screen feature but I wonder, is there a way to print immediate a user provider static string (text, in the console)? E.g. just identifying the program and letting the user know what it will take a few seconds to start.
> 
> Alternatively, does pyinstaller provide any use feedback during the app initialization such as a progress bar or any other textual indication?

No. If you want your onefile application to print a message at the start (e.g., before it starts unpacking itself), you'll need to modify the bootloader by adding a `printf()` to corresponding place. Then rebuild the bootloader, install the modified version, and rebuild your application with it.

**Reply by zapta** [reply]
For the record, I just filed this feature request.

https://github.com/pyinstaller/pyinstaller/issues/7511

**Comment by bwoodsend** [comment]
In all honesty, if startup time is getting uncomfortably long, you probably shouldn't be using onefile mode.

**Comment by zapta** [comment]
What is the alternative, how does a distribution of a non onefile app looks
like?  A zip with a bunch of files and an exe (e.g. on windows)?

**Reply by bwoodsend** [reply]
Either that or use [innosetup](https://jrsoftware.org/isinfo.php) to create an installer from a one-dir build.
